core/src/main/scala/kafka/raft/KafkaMetadataLog.scala

```diff
@@ -294,19 +298,14 @@ final class KafkaMetadataLog private (
     }
   }
 
-  override def deleteBeforeSnapshot(logStartSnapshotId: OffsetAndEpoch): Boolean = {
+  override def deleteBeforeSnapshot(deleteBeforeSnapshotId: OffsetAndEpoch): Boolean = {
```

Comment:
Since this is no longer used to increase the Log Start Offset, I think this method can go away.